returns/result.py

```diff
@@ -119,9 +119,9 @@ def bind(
         self,
         function: Callable[
             [_ValueType],
-            Kind2['Result', _NewValueType, _ErrorType],
+            Kind2['Result', _NewValueType, _ErrorType | _NewErrorType],
```

Comment:
The problem with this approach is that it works with `Exception` subtypes, but it does not work if you have `Result[int, int]` and `Result[int, str]`

How would you compose these two objects? `Result[int, int | str]` might not make any sense.

Reply:
> How would you compose these two objects? `Result[int, int | str]` might not make any sense.
Thank you for the review! I see your point, but isn’t `int | str` better than dealing with `Any`? At least with `int | str`, we can use pattern matching to handle each case explicitly.